File: 733. Flood Fill/floodFill.java
class floodFill {
    public int[][] solution(int[][] image, int sr, int sc, int newColor) {
    	int oldColor = image[sr][sc];
    	dfs(image, sr, sc, newColor, oldColor);
    	return image;
    }

    public void dfs(int[][] image, int i, int j, int newColor, int oldColor) {
    	if (i < 0 || i >= image.length || j < 0 || j >= image[0].length || image[i][j] == newColor || image[i][j] != oldColor) {
    		return;
    	}
    	else {
    		image[i][j] = newColor;
    		dfs(image, i-1, j, newColor, oldColor);
    		dfs(image, i+1, j, newColor, oldColor);
    		dfs(image, i, j-1, newColor, oldColor);
    		dfs(image, i, j+1, newColor, oldColor);
    	}
    }

    public static void main(String[] argv) {
    	floodFill obj = new floodFill();
    	int[][] img = {{1,1,1},{1,1,0},{1,0,1}};
    	int r = 1, c = 1, nColor = 2;
    	int[][] res = obj.solution(img, r, c, nColor);
    	for (int i=0; i < res.length; i++) {
    		for (int j=0; j < res[0].length; j++) {
    			System.out.print(res[i][j] + " ");
    		}
    		System.out.println(" ");
    	}
    }
}

File: 733. Flood Fill/floodFill.py
import queue

class floodFill:
    def solution(self, image, sr, sc, newColor):
        oldColor = image[sr][sc]
        if oldColor == newColor:
            return image
        dx = [-1, 1, 0, 0]
        dy = [0, 0, -1, 1]
        q = queue.Queue()
        q.put((sr,sc))
        width = len(image)
        length = len(image[0])
        while not q.empty():
            temp = q.get()
            x = temp[0]
            y = temp[1]
            image[x][y] = newColor
            for i in range(0, 4):
                r = x + dx[i]
                c = y + dy[i]
                if 0 <= r < width and 0 <= c < length and image[r][c] == oldColor:
                    q.put((r, c))
        return image

# img = [[1,1,1],[1,1,0],[1,0,1]]
img = [[0,0,0],[0,1,1]]
sr, sc, newColor = 1, 1, 1
obj = floodFill()
res = obj.solution(img, sr, sc, newColor)
print(res)